### DISCUSSION TITLE
Useful resources for Mirth Connect

### DISCUSSION THREAD

**Discussion by mjtrac** [nextgenhealthcare/connect#5760] [General]
"pacmano" provided this link in response to a specific question, but it may be of general value to people working with Mirth Connect: https://github.com/nextgenhealthcare/connect-examples

If you know of other resources, please consider posting them here.

**Comment by mjtrac** [comment]
Here is the specification for the Javascript extension E4X (Ecmascript for XML) used in Mirth Connect filters and transformers to work with XML: https://www.ecma-international.org/wp-content/uploads/ECMA-357_2nd_edition_december_2005.pdf

And here is a "quickstart": https://svn.wso2.org/repos/wso2/tags/carbon/0.1alpha/mashup/java/xdocs/e4xquickstart.html

If you are looking for where certain Javascript functions are defined before your Javascript is called, you can find them here in the github source tree by searching for createSegment and then clicking on the link to the file JavaScriptBuilder.java

**Comment by jonbartels** [comment]
https://github.com/stars/rogin/lists/mirth-related rogin (Richard on Slack) has curated a fantastic list of other repos

Loyd Bittle started the MC Users Group and listed several projects - https://www.mcug.org/opensourceprojects

**Reply by rjkroll** [reply]
Thanks for sharing these two links.   I never knew about them!
